src/iuse.h

```diff
@@ -311,8 +311,22 @@ struct use_function {
     void operator=(const use_function &other);
 
     bool operator==(use_function f) const {
-        return function_type == USE_FUNCTION_CPP && f.function_type == USE_FUNCTION_CPP &&
-        f.cpp_function == cpp_function;
+        if( function_type != f.function_type ) {
+            return false;
+        }
+
+        switch( function_type ) {
+            case USE_FUNCTION_NONE:
+                return true;
+            case USE_FUNCTION_CPP:
+                return f.cpp_function == cpp_function;
+            case USE_FUNCTION_ACTOR_PTR:
+                return f.actor_ptr == actor_ptr;
```

Comment:
That is probably useless, each `use_function` object has it's own actor pointer, they are *not* shared. Look at the copy constructor of `use_function`, it stores a new pointer (created by calling `iuse_actor::clone`).
```C++
use_function a( some_iuse_actor_ptr );
use_function b( a );
if( a == b ) ... // will be false
if( a.get_actor_ptr() == b.get_actor_ptr() ) ... // will be false
```